File: web/src/main/java/top/bettercode/summer/web/serializer/annotation/JsonDivision.java
package top.bettercode.summer.web.serializer.annotation;

import com.fasterxml.jackson.annotation.JacksonAnnotationsInside;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import top.bettercode.summer.web.serializer.DivisionSerializer;

@Target({ElementType.METHOD, ElementType.FIELD})
@Retention(RetentionPolicy.RUNTIME)
@JacksonAnnotationsInside
@JsonSerialize(using = DivisionSerializer.class)
public @interface JsonDivision {

  /**
   * @return 是否使用虚拟节点
   */
  boolean value() default false;

}

File: web/src/main/kotlin/top/bettercode/summer/web/serializer/annotation/JsonDivision.kt
package top.bettercode.summer.web.serializer.annotation

import com.fasterxml.jackson.annotation.JacksonAnnotationsInside
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import top.bettercode.summer.web.serializer.DivisionSerializer

@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER, AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
@JacksonAnnotationsInside
@JsonSerialize(using = DivisionSerializer::class)
annotation class JsonDivision(
        /**
         * @return 是否使用虚拟节点
         */
        val value: Boolean = false)
